tests/integration/conftest.py

```diff
@@ -439,3 +439,113 @@ async def expected_dashboard_titles():
         dashboard = json.loads(path.read_text())
         titles.append(dashboard["title"])
     return titles
+
+
+@pytest_asyncio.fixture(scope="module")
+async def prometheus_app(ops_test, k8s_model):
+    _, k8s_alias = k8s_model
+    with ops_test.model_context(k8s_alias) as m:
+        log.info("Deploying prometheus-k8s ...")
+        await m.deploy(entity_url="prometheus-k8s", trust=True, channel="edge")
+
+        await m.block_until(lambda: "prometheus-k8s" in m.applications, timeout=60 * 3)
+        await m.wait_for_idle(status="active")
```

Comment:
what if this was 
```suggestion
        await m.wait_for_idle(status="active", timeout=60 * 3)
```

or even this if you REALLY only want prometheus stable

```suggestion
        await m.wait_for_idle(apps=["prometheus-k8s"], status="active", timeout=60 * 3)
```

and finally, if this charm is going into `error` state for a little bit you can do this:

```suggestion
        await m.wait_for_idle(apps=["prometheus-k8s"], status="active", timeout=60 * 3, raise_on_error=False)
```